### ISSUE TITLE
Videos sans son

### ISSUE THREAD

**Issue by mathieugouin** [service-paradis/plugin.tva.videos#2]
Juste par curiosité, quel est ton système qui roule kodi?

Je suis sur un raspberry pi 3 et je suis capable de voir les vidéos mais je n'ai pas de son.

Avec la touche 'o' je vois qu'il n'y a pas de stream audio.

**Comment by service-paradis** [comment]
Salut,

Je suis aussi sur un raspberry pi 3 avec Kodi 17.

J'ai bel et bien sons et images de mon côté. Peux-tu me dire quels vidéos tu as essayés? Je vais essayer de reproduire le problème.

**Comment by mathieugouin** [comment]
Séries, Boomerang, A voir et revoir saison 3, épisode du 6 novembre 2017.

J'ai essayé blue moon aussi et même résultat.

Ça semble généralisé...

**Comment by service-paradis** [comment]
Merci du suivi, ça fonctionne de mon côté.

J'ai aussi essayé Kodi 17 sur un système Windows et sur Arch Linux. Ça fonctionne de mon côté.

Je vais faire des tests avec Kodi 16 quand j'aurai une minute!

**Comment by service-paradis** [comment]
@mathieugouin 
J'ai peut-être une piste. Actuellement, j'ai le choix entre plusieurs source pour jouer les épisodes. Je choisissais la meilleure qualité disponible qui s'adonne être MPEG avec le son en AAC. 
Je serais tenter d'essayer une source vidéo MP4 avec le codec H264 pour voir si ça pourrait corriger ton problème.
Si je fais le changement, tu serais prêt à l'essayer et me donner ton feedback?

**Comment by mathieugouin** [comment]
Oui, pas de problème, je peux te tester ça volontiers.  Le mieux, fais-moi une branche au cas où ça fonctionne pas.

**Comment by service-paradis** [comment]
@mathieugouin 
Tu devras probablement refaire quelques ajustements pour Kodi 16, mais j'ai modifié le code pour le stream vidéo MP4 / H264 (https://github.com/service-paradis/plugin.tva.videos/tree/MP4_H264).

En espérant que ça fonctionne! J'attends de tes nouvelles.

**Comment by mathieugouin** [comment]
@service-paradis
Je confirme que ça fonctionne avec H264 :)

**Comment by mathieugouin** [comment]
Un autre point de curiosité, quel est ton setup?  Moi, c'est Pi3 => TV (qui date un peu...) 

Je me demandais si par hasard le mode qui n'est pas supporté de mon côté l'est du tiens à cause d'un ampli cinéma maison qui serait capable de le décoder?  Ex: Pi3 => Ampli => TV

J'ai trouvé un setting aussi dans Kodi qui active passthrough pour l'audio, j'ai essayé divers configuration sans succès avec la branche master.

**Comment by mathieugouin** [comment]
Une autre idée de même, dis-moi ce que tu en penses, on pourrait mettre ça dans les settings de l'addon pour choisir quel format audio/video?

Merci encore :)

**Comment by service-paradis** [comment]
@mathieugouin 
Content de voir que ça fonctionne! Mon setup est similaire au tien. Pi3 branché directement sur la télé.

J'ai l'impression que Kodi 16 et Kodi 17 gère différemment le format m3u8.

C'est une bonne idée de mettre le format dans les settings. J'opterais pour MP4/H264 par défaut pour la compatibilité qui pourrait être changé pour m3u8 dans les settings. Je pourrais peut-être aussi chargé le format selon la version de Kodi.

Il faut que je me familiarise un peu plus avec le fonctionnement des settings dans Kodi!

Merci pour ton input!
